include/os/linux/kernel/linux/vmalloc_compat.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2020, Sebastian Gottschall / NewMedia-NET GmbH. All rights reserved.
+ * Copyright (c) 2020, Michael Niewöhner. All rights reserved.
+ */
+
+#ifndef _ZFS_VMALLOC_H
+#define	_ZFS_VMALLOC_H
+
+#include <linux/vmalloc.h>
+
+/*
+ * 5.8 API change
+ */
+
+#ifdef HAVE_VMALLOC_PAGE_KERNEL
+#define	__vmalloc(size, gfp_flags) \
+    __vmalloc(size, gfp_flags, PAGE_KERNEL)
+#endif /* HAVE_VMALLOC_PAGE_KERNEL */
```

Comment:
Since we only intend to use this wrapper from the SPL, let's move it to `include/os/linux/spl/sys/kmem.h` with the other Linux kmem compatibility code and add the `spl_` prefix.  I was thinking something like this after the `void *spl_kvmalloc(size_t size, gfp_t flags);` prototype.


```c
/*
 * 5.8 API change, pgprot_t argument removed.
 */
#ifdef HAVE_VMALLOC_PAGE_KERNEL
#define spl_vmalloc(size, flags)        __vmalloc(size, flags, PAGE_KERNEL)
#else
#define spl_vmalloc(size, flags)        __vmalloc(size, flags)
#endif
```

Reply:
ack, we can even move the __GFP_RETRY_MAYFAIL compat code there

Reply:
@c0d3z3r0 yep. makes sense

Reply:
@behlendorf include/os/linux/spl/sys is for userspace and kernel space. but include/os/linux/kernel/linux  is for kernel only. so kernel wrappers should be placed in this folder as i initially did. spl_vmalloc should never be used from userspace and should be out of reach by its declaration